chain/bitcoind_client.go

```diff
@@ -769,16 +789,20 @@ func (c *BitcoindClient) onBlockDisconnected(hash *chainhash.Hash, height int32,
 	timestamp time.Time) {
 
 	if c.shouldNotifyBlocks() {
-		select {
-		case c.notificationQueue.ChanIn() <- BlockDisconnected{
+		n := BlockDisconnected{
 			Block: wtxmgr.Block{
 				Hash:   *hash,
 				Height: height,
 			},
 			Time: timestamp,
-		}:
+		}
+		select {
+		case c.notificationQueue.ChanIn() <- n:
 		case <-c.quit:
 		}
+		if c.publicNotificationQueue != nil {
```

Comment:
why not have it inside the case block? otherwise it may run after quit
applies to all the other such cases